smolyay/basis.py

```diff
@@ -57,6 +61,23 @@ def derivative(self, x):
         """
         pass
 
+    def outside_domain(self, x):
+        """Check if the input is within the natural domain.
+
+        Parameters
+        ----------
+        x : float, numpy:ndarray
+            One-dimensional points.
+
+        Returns
+        -------
+        bool
+            True if input was outside domain, False otherwise
+        """
+        return numpy.any(numpy.greater(x, self.natural_domain[1])) or numpy.any(
+            numpy.less(x, self.natural_domain[0])
```

Comment:
I suggest this is done element wise, and allow the caller to decide whether they take `all` or not. These should also be >= and <= since the domains are defined to be closed intervals.
```suggestion
        return numpy.logical_and(numpy.greater_equal(x, self.domain[0]),
            numpy.less_equal(x, self.domain[1]))
```

If you think it would be convenient, you could add a method like:
```py
def check_all_in_domain(self, x):
    return numpy.all(self.check_in_domain(x))
```